pkg/rbac/cacheMetadata.go

```diff
@@ -0,0 +1,34 @@
+// Copyright Contributors to the Open Cluster Management project
+package rbac
+
+import (
+	"sync"
+	"time"
+
+	"github.com/stolostron/search-v2-api/pkg/config"
+)
+
+// Common fields to manage a cached data field.
+type cacheMetadata struct {
+	err       error         // Error while retrieving the data from external API.
+	lock      sync.Mutex    // Locks the data field while requesting the latest data.
+	updatedAt time.Time     // Time when the data field was last updated.
+	ttl       time.Duration // Time-to-live, time duration for which this cache is valid.
```

Comment:
will ttl be different for different caches ( namespace vs clusterscoped vs ..)going forward?

Reply:
Assuming this field will be set in the upcoming PRs.

Reply:
is the idea that we are moving it out of config.go or will this point to it?

Reply:
currently, default ttl values differ for token review and shared/user data cache. But we still need to adjust those times to something more realistic.

Reply:
Right, ttl is intended to be adjusted for different type of data (token review, shared, user).  This PR is limiting the scope to sharedData, but it will get more use in my next change.